### PR TITLE
set `?lang=` when changing tabs in TS docs and app dev guide

### LINKED ISSUE #1271: [ Feature request ] Update ?lang= query string on click
## Required

**What is the issue?**

When I visit:

https://docs.temporal.io/application-development-guide/

Select a section, select Java, and share the link, the recipient sees Go code.

**What is the expected behavior?**

The recipient sees Java code.

## Solution

If the person knows about #1261, they can add `?lang=java` to the URL. Most don't. A solution for them would be:

- When a tab is clicked, the URL query string updates.

This feature *might* be implemented upstream in future as part of https://github.com/facebook/docusaurus/issues/7008

### PR DESCRIPTION
**Pull request by vkarpov15**
Fix #1271

## What does this PR do?

One potential fix for #1271: swizzle the Tab component to make `handleTabChange()` trigger a `replaceState()`. Not as elegant as I'd like, but there doesn't seem to be any other way to plug into tab change behavior in docusaurus. On the bright side, combined with `set-tab-language.js`, now both the app dev guide and TS docs add a `?lang=ts` or `?lang=java` when you click on a tab.

### CODE REVIEW

File: static/scripts/set-tab-language.js [hunk 1]
```diff
@@ -13,9 +13,9 @@
       return entries;
     }, {});
 
-  if (entries.language === "js") {
+  if (entries.lang === "js") {
     window.localStorage.setItem("docusaurus.tab.language", "js");
-  } else if (entries.language === "ts") {
+  } else if (entries.lang === "ts") {
     window.localStorage.setItem("docusaurus.tab.language", "ts");
   }
```
Comment (lorensr) [inline static/scripts/set-tab-language.js:20]: If it's `js` or `ts`, we also want to set `docusaurus.tab.site-lang` to `typescript`

**Review by lorensr — APPROVED** [review]

File: docusaurus.config.js [hunk 2]
```diff
@@ -344,6 +344,11 @@ module.exports = {
       async: true,
       defer: true,
     },
+    {
+      src: "/scripts/set-tab-language.js",
+      async: true,
+      defer: true,
+    },
```
[the remark this answers is not in the document]
Comment (vkarpov15) [reply docusaurus.config.js:351]: Looks like it was deleted in https://github.com/temporalio/documentation/commit/4a7d077c835f50a3bedbc7d1e82c46277efcbb00 . Not sure why, maybe merge conflict?

[the remark this answers is not in the document]
Comment (lorensr) [reply docusaurus.config.js:351]: @djmagee @rachfop Guessing that was a mistake?

**Comment by cretz** [comment]
Can you confirm how this affects URL fragments to headings?

Like if I am defaulted to Go and I open a new browser window visit `/application-development-guide/?lang-java#start-activity-execution`, what happens? Does this first move to that anchor as expected and _then_ switch the Java tab? Does this cause any jitter?

**Review by djmagee — APPROVED** [review]

**Comment by cretz** [comment]
> Any way to make it append the url param to the end of the url - after the anchor - rather than clearing it?

I know this is totally doable btw, as I have done it plenty in the past. Granted it's before the anchor. Basically you inject the query param and put the URL back together w/ its fragment.

**Comment by flossypurse** [comment]
Need this now to fix functionality - we can close gaps in feature after.

**Pull request merged**